plot-geotiff-pixel-bbox.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+###########################################################
+# Plots a GeoTiff with a pixel count bounding box overlay.
+# Use to determine pixel values for ground control points
+# when geo-referencing or creating/testing cutlines.
+#######################################################
+import argparse
+import logging
+import os
+import rasterio
+from rasterio import windows
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle
+
+logging.basicConfig(format='%(message)s', level=logging.INFO)
+logger = logging.getLogger()
+
+
+def run(args):
+    src_file = args.src
+    min_x = args.min_x
+    max_x = args.max_x
+    min_y = args.min_y
+    max_y = args.max_y
+
+    logging.info('Reading band 1 from ' + src_file)
+    raster_src = rasterio.open(src_file)
+
+    logging.info('Building overlay')
+    slice_raster = (slice(min_y, max_y), slice(min_x, max_x))
+    window_slice = windows.Window.from_slices(*slice_raster)
+
+    plt.imshow(raster_src.read(1))
+    plt.title(os.path.basename(src_file))
+    ax = plt.gca()
+    ax.add_patch(
+        Rectangle(
+            (window_slice.col_off, window_slice.row_off),
+            width=window_slice.width,
+            height=window_slice.height,
+            fill=True,
+            alpha=.5,
+            color="red"
+        )
+    )
+
+    plt.show()
```

Comment:
Most systems with Python installed will allow you to use `plt.show()` from a terminal, but in certain circumstances (e.g. SSH'ed into a server or something) you likely won't be able to do that (there are some steps required to actually get "windows" from a server to a local computer). 

Another option is to replace `plt.show()` with `plt.savefig(filename)`. This also allows you to set the DPI of the output more easily (e.g. `plt.savefig(filename, dpi=300)`) to get higher-resolution outputs if that's useful. 

If you want to do this you could do an optional parameter with a default (e.g.defaults to output.png with an option to add`-o my_custom_filename.png`).